lib/msal-node/src/client/ClientAssertion.ts

```diff
@@ -109,9 +112,25 @@ export class ClientAssertion {
 
         const header: jwt.JwtHeader = {
             alg: JwtConstants.RSA_256,
-            x5t: EncodingUtils.base64EncodeUrl(this.thumbprint, "hex"),
         };
 
+        if (this.thumbprint256) {
+            Object.assign(header, {
+                "x5t#S256": EncodingUtils.base64EncodeUrl(
+                    this.thumbprint256,
+                    "hex"
+                ),
+            } as Partial<jwt.JwtHeader>);
+        } else {
+            Object.assign(header, {
+                x5t: EncodingUtils.base64EncodeUrl(
+                    // at this point, due to prior error checking, thumbprint will be guaranteed to be a string here
+                    this.thumbprint as string,
```

Comment:
nit: why does `thumbprint` need to be cast to string but `thumbprint256` does not?

Reply:
Because the linter was showing an error stating that thumbprint could be undefined, since it's an optional parameter. However, I already have error checking in `Configuration.ts` that makes sure at least one of `thumbprint` and `thumbprint256` is defined. I need to cast to string here to avoid the linter error. As the comment says, "at this point, due to prior error checking, thumbprint will be guaranteed to be a string here".

Reply:
addressed this with code improvement